pkg/util/metrics_helper_test.go

```diff
@@ -999,6 +1000,56 @@ func uint64p(v uint64) *uint64 {
 	return &v
 }
 
+func BenchmarkGetLabels_SmallSet(b *testing.B) {
+	m := prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "test",
+		ConstLabels: map[string]string{
+			"cluster": "abc",
+		},
+	}, []string{"reason", "user"})
+
+	m.WithLabelValues("bad", "user1").Inc()
+	m.WithLabelValues("worse", "user1").Inc()
+	m.WithLabelValues("worst", "user1").Inc()
+
+	m.WithLabelValues("bad", "user2").Inc()
+	m.WithLabelValues("worst", "user2").Inc()
+
+	m.WithLabelValues("worst", "user3").Inc()
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		GetLabels(m, map[string]string{"user": "user1", "reason": "worse"})
```

Comment:
The linter wants you to check the return error. To have you negative perf impact you can do:

```suggestion
		if err := GetLabels(m, map[string]string{"user": "user1", "reason": "worse"}); err != nil {
		    b.Fatal(err.Error())
		}
```

Reply:
updated in new commit. thanks